api/millenium_patron.py

```diff
@@ -245,7 +246,6 @@ def authenticated_patron(self, db, header):
                 # Sync our internal Patron record with what the API
                 # says.
                 self.update_patron(patron, identifier)
-                patron.last_external_sync = now
```

Comment:
Setting last_external_sync here instead of in update_patron ensures that a patron gets synced the first time we hear about them (when we use get_one_or_create), and then they also get synced on their second request.